server/k8s.go

```diff
@@ -281,3 +283,41 @@ func (w *k8sWatcherImpl) buildScaleUpFunction(service *core.Service) func(ctx co
 		return nil
 	}
 }
+
+func (w *k8sWatcherImpl) buildScaleDownFunction(service *core.Service) func(ctx context.Context) error {
```

Comment:
This and `buildScaleUpFunction` can probably be refactored into a combined method I think. I'm thinking they're main difference would be via `from` and `to` arguments to specify which way the scaling is going.